plugins/module_utils/s3.py

```diff
@@ -22,33 +28,49 @@
 import string
 
 
+def s3_head_objects(client, parts, bucket, obj, versionId):
+    args = {"Bucket": bucket, "Key": obj}
+    if versionId:
+        args["VersionId"] = versionId
+
+    for part in range(1, parts + 1):
+        args["PartNumber"] = part
+        yield client.head_object(**args)
+
+
+def calculate_checksum_with_file(client, parts, bucket, obj, versionId, filename):
```

Comment:
Note for the future, It would be nice to extend this to cover the other algorithms https://aws.amazon.com/blogs/aws/new-additional-checksum-algorithms-for-amazon-s3/